Course/Course/Presentation/Dates/CourseDatesViewModel.swift

```diff
@@ -91,4 +94,49 @@ public class CourseDatesViewModel: ObservableObject {
             errorMessage = CourseLocalization.Error.componentNotFount
         }
     }
+    
+    @MainActor
+    func shiftDueDates(courseID: String, withProgress: Bool = true) async {
+        isShowProgress = withProgress
+        dueDatesShifted = false
+        do {
+            try await interactor.shiftDueDates(courseID: courseID)
+            NotificationCenter.default.post(name: .shiftCourseDates, object: courseID)
+            isShowProgress = false
+        } catch let error {
+            isShowProgress = false
+            if error.isInternetError || error is NoCachedDataError {
+                errorMessage = CoreLocalization.Error.slowOrNoInternetConnection
+            } else {
+                errorMessage = CoreLocalization.Error.unknownError
+            }
+        }
+    }
+    
+    deinit {
+        let center = NotificationCenter.default
+        guard let shiftCourseDatesObserver = self.shiftCourseDatesObserver else { return }
+        center.removeObserver(shiftCourseDatesObserver)
+    }
+}
+
+extension CourseDatesViewModel {
+    func addNotificationObserver() {
```

Comment:
It would be nice if you can rename it to `addObservers` so in future other observes will also be added to the same method.